pkg/probe/grpc/grpc_test.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpc
+
+import (
+	"context"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	grpchealth "google.golang.org/grpc/health/grpc_health_v1"
+
+	"k8s.io/kubernetes/pkg/probe"
+)
+
+func TestNew(t *testing.T) {
+	t.Run("Should: implement Probe interface", func(t *testing.T) {
+		s := New()
+		assert.Implements(t, (*Prober)(nil), s)
+	})
+}
+
+type successServerMock struct {
+}
+
+func (s successServerMock) Check(context.Context, *grpchealth.HealthCheckRequest) (*grpchealth.HealthCheckResponse, error) {
+	return &grpchealth.HealthCheckResponse{
+		Status: grpchealth.HealthCheckResponse_SERVING,
+	}, nil
+}
+
+func (s successServerMock) Watch(_ *grpchealth.HealthCheckRequest, stream grpchealth.Health_WatchServer) error {
+	return stream.Send(&grpchealth.HealthCheckResponse{
+		Status: grpchealth.HealthCheckResponse_SERVING,
+	})
+}
+
+type errorTimeoutServerMock struct {
+}
+
+func (e errorTimeoutServerMock) Check(context.Context, *grpchealth.HealthCheckRequest) (*grpchealth.HealthCheckResponse, error) {
+	time.Sleep(time.Second * 4)
```

Comment:
can we get rid of the sleeps and use some active loop for detecting the starts conditions?
this is a source of flakiness on busy CIs

Reply:
OK,  how to implment this.Http probe unit test seems like also sleep some sceonds in here https://github.com/kubernetes/kubernetes/blob/master/pkg/probe/http/http_test.go#L303